components/hab/src/command/studio/enter.rs

```diff
@@ -13,12 +13,39 @@ use crate::{config,
 
 pub const ARTIFACT_PATH_ENVVAR: &str = "ARTIFACT_PATH";
 
+const AUTH_TOKEN_ENVVAR: &str = "HAB_AUTH_TOKEN";
+const BLDR_URL_ENVVAR: &str = "HAB_BLDR_URL";
+const CTL_SECRET_ENVVAR: &str= "HAB_CTL_SECRET";
 const ORIGIN_ENVVAR: &str = "HAB_ORIGIN";
 const STUDIO_CMD: &str = "hab-studio";
 const STUDIO_CMD_ENVVAR: &str = "HAB_STUDIO_BINARY";
 const STUDIO_PACKAGE_IDENT: &str = "core/hab-studio";
 
 pub fn start(ui: &mut UI, args: &[OsString]) -> Result<()> {
+    if henv::var(AUTH_TOKEN_ENVVAR).is_err() {
+        let config = config::load()?;
```

Comment:
I debated on this one here...  I thought this might have been done for a reason.  

Potentially, if all of the env vars are set, you never need to call `config::load()`, but if none of the environment variables are set, we're doing a `config::load()` four times...  Does rust cache those reads, or does that cause four reads?

Reply:
They won't be cached. I had the same thought about the fact that `config::load` may not be called at all and whether it made sense to change it to always be called, but I think that's the best approach. It shouldn't be expensive, and in the rare case that it causes an error (`cli.toml` is corrupt, or there are permissions problems or something), I think it's easier to track down if we *always* call `config::load`.

Reply:
Cool, so along the same lines of thinking :+1: